_posts/2021-04-17-dd_00001_intro.md

```diff
@@ -0,0 +1,324 @@
+operations have their foundation in the theory of **Boolean Algebra**. That now
+means, we can design logic circuits that can compute stuff. Well, there you have
+it: the processor inside a _computer_ is nothing but that!
+
+Now let's see how we design such logic circuits.
+
+## EDA, HDLs and Testing
+
+In the early days of microchips, the logic gate circuits were designed mostly by
+hand. This was kind of doable, because the number of transistors compared to
+today was much smaller, usually between 100s - 1000s. However, as the
+transistors were getting smaller and the chips more complex, engineers realized
+that designing those chips by hand would take a lot of time and was prone to
+errors. This was when the era of **Electronic Design Automation** (EDA) began.
+Engineers would now design their chips using so-called **Hardware Description
+Languages** (HDLs). These HDLs were inspired by programming languages used for
+writing software (however, writing software and writing digital hardware are two
+entirely different things!). The HDL code of the design is then _translated_
+into an equivalent logic gates description of the circuit by an _EDA software
+tool_. Again, this has some similarities with modern software design, where a
+_compiler_ translates the higher-level code into assembly language for the CPU.
+
+With the use of HDLs, the engineers didn't need to focus on each and every
+transistor anymore and could put more time and effort into the actual design of
+their system. Two HDLs emerged in the 1980s: VHDL and Verilog. Those are still
+the main languages to use when it comes to digital design. In the last couple of
+decades, new trends in digital design emerged, among which the most popular was
+**High-Level Synthesis** (HLS). The idea of HLS is to take an algorithm written
+in some high-level programming language, such as C, and automatically generate
+an optimized digital circuit for that particular algorithm. To this date,
+researchers all over the world are trying to come up with new languages and
+design paradigms with the hope of eventually replacing the traditional HDLs.
+
+The rapid advancement of microchips over the past decades was - apart from the
+technological improvements in transistors - made possible by EDA. EDA tools
+automate many time-consuming tasks in chip design. Two of the most important
+tasks are _Synthesis_ and _Place & Route_. In Synthesis, the HDL code is
+translated into a logic gates circuit (not a physical one, but described in some
+file). Then, in Place&Route, the EDA tool tries to find the optimal location for
+each gate on the final chip. Both tasks, try to follow certain _constraints_
+given by the engineer, such as the final performance or power consumption of the
+chip. The tools use physical models of the actual gates to come up with
+estimations on performance or power consumption. The engineer can then analyze
+the results, and refine his design if he is not satisfied.
```

Comment:
I think this part is very good but is a lot of knowledge for a beginner. Maybe again a note that you will cover it step by step in the upcoming posts?

Reply:
d9f3bbd

better?